pkg/registry/networking/networkpolicy/storage/storage.go

```diff
@@ -34,24 +39,29 @@ type REST struct {
 }
 
 // NewREST returns a RESTStorage object that will work against NetworkPolicies.
-func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, error) {
+func NewREST(optsGetter generic.RESTOptionsGetter) (*REST, *StatusREST, error) {
 	store := &genericregistry.Store{
 		NewFunc:                  func() runtime.Object { return &networkingapi.NetworkPolicy{} },
 		NewListFunc:              func() runtime.Object { return &networkingapi.NetworkPolicyList{} },
 		DefaultQualifiedResource: networkingapi.Resource("networkpolicies"),
 
-		CreateStrategy: networkpolicy.Strategy,
-		UpdateStrategy: networkpolicy.Strategy,
-		DeleteStrategy: networkpolicy.Strategy,
+		CreateStrategy:      networkpolicy.Strategy,
+		UpdateStrategy:      networkpolicy.Strategy,
+		DeleteStrategy:      networkpolicy.Strategy,
+		ResetFieldsStrategy: networkpolicy.Strategy,
 
 		TableConvertor: printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(printersinternal.AddHandlers)},
 	}
 	options := &generic.StoreOptions{RESTOptions: optsGetter}
 	if err := store.CompleteWithOptions(options); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
-	return &REST{store}, nil
+	statusStore := *store
```

Comment:
Should this be gated?  It seems like this is the actual functionality.

Reply:
if someone writes the status and then disables the gate, that would not register the endpoint, which means they could never clear it.  Cancel this comment.